salt/modules/state.py

```diff
@@ -2343,9 +2345,9 @@ def event(tagmatch='*',
                 )
                 sys.stdout.flush()
 
-            count -= 1
-            log.debug('Remaining event matches: %s', count)
-
+            if count > -1:
+                count -= 1
+                log.debug('Remaining event matches: %s', count)
```

Comment:
This is a good change, it prevents a bunch of confusing debugging messages that I've seen from time to time as `state.event` is used.